src/osdc/Objecter.cc

```diff
@@ -451,12 +451,12 @@ void Objecter::shutdown()
     cop->put();
   }
 
-  if (tick_event) {
-    if (timer.cancel_event(tick_event)) {
+  if (tick_event.read()) {
+    if (timer.cancel_event(tick_event.read())) {
```

Comment:
Hmm.  We're holding Objecter::rwlock here, and the timer is potentially in the Objecter::tick callback, blocking on acquiring rwlock.  If we cancel the event here, that's a no-op because timer has already removed the event from the queue.  We'll go ahead and zero it, and eventually give up rwlock.  Objecter::tick will acquire rwlock, and immediately assert out on the condition that tick_event is nonzero.  Probably the assertion inside Objecter::tick needs to move the other side of the `initialized` check so that it never gets hit in this case.